File: app/src/main/java/com/example/museumapplication/ui/auth/SignupActivity.java
package com.example.museumapplication.ui.auth;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.example.museumapplication.R;
import com.example.museumapplication.utils.AuthProviders.IBaseAuth;
import com.example.museumapplication.utils.AuthUtils;
import com.example.museumapplication.utils.AuthProviders.EmailAuth;

public class SignupActivity extends AppCompatActivity {

    IBaseAuth emailAuth;
    EditText email;
    EditText password;
    EditText repeatPass;
    EditText name;
    EditText verificationCode;

    Button requestCodeButton;
    TextView timerText;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_signup);

        email=  findViewById(R.id.emailEditText);
        password = findViewById(R.id.passwordEditText);
        repeatPass= findViewById(R.id.repeatPassEditText);
        name = findViewById(R.id.nameEditText);
        verificationCode = findViewById(R.id.verificationCodeEditText);

        requestCodeButton = findViewById(R.id.requestCodeButton);
        timerText = findViewById(R.id.timerText);

        emailAuth = new EmailAuth();

    }

    public void requestVerificationButtonClicked(View view) {
        ((EmailAuth)emailAuth).createVerificationCode(this);
    }

    public void registerButtonClicked(View view) {

        if(AuthUtils.checkFields(email, password, repeatPass,verificationCode,name)){
            ((EmailAuth)emailAuth).setCredentialInfo(email.getText().toString(), password.getText().toString(), verificationCode.getText().toString(), this);
            ((EmailAuth)emailAuth).register();
        }
    }


}

File: app/src/main/java/com/example/museumapplication/ui/auth/SignupActivity.kt
package com.example.museumapplication.ui.auth

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.museumapplication.R
import com.example.museumapplication.utils.auth.EmailAuth
import com.example.museumapplication.utils.auth.IBaseAuth
import com.example.museumapplication.utils.auth.AuthUtils.checkFields

class SignupActivity : AppCompatActivity() {
    var emailAuth: IBaseAuth? = null
    var email: EditText? = null
    var password: EditText? = null
    var repeatPass: EditText? = null
    var name: EditText? = null
    var verificationCode: EditText? = null
    var requestCodeButton: Button? = null
    var timerText: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_signup)
        email = findViewById(R.id.emailEditText)
        password = findViewById(R.id.passwordEditText)
        repeatPass = findViewById(R.id.repeatPassEditText)
        name = findViewById(R.id.nameEditText)
        verificationCode = findViewById(R.id.verificationCodeEditText)
        requestCodeButton = findViewById(R.id.requestCodeButton)
        timerText = findViewById(R.id.timerText)
        emailAuth = EmailAuth()
    }

    fun requestVerificationButtonClicked(view: View?) {
        (emailAuth as EmailAuth?)!!.createVerificationCode(this)
    }

    fun registerButtonClicked(view: View?) {
        if (checkFields(email!!, password!!, repeatPass!!, verificationCode!!, name!!)) {
            (emailAuth as EmailAuth?)!!.setCredentialInfo(email!!.text.toString(), password!!.text.toString(), verificationCode!!.text.toString(), name!!.text.toString(), this)
            (emailAuth as EmailAuth?)!!.register()
        }
    }
}